lib/api.js

```diff
@@ -1,48 +1,77 @@
 const { config } = require('./config');
 const request = require('request-promise').defaults({
     method: 'POST',
-    baseUrl: 'https://apptoogoodtogo.com/index.php/api_tgtg',
-    json: true,
+    baseUrl: 'https://apptoogoodtogo.com/api',
     headers: config.get('api.headers')
 });
 const session = {};
 
 module.exports = {
     login,
+    refreshToken,
     listFavoriteBusinesses
 };
 
 function login(credentials = config.get('api.credentials')){
     return request({
-        url: '/login',
-        form: { 
-            email: credentials.email, 
-            password: credentials.password 
+        url: '/auth/v1/loginByEmail',
+        json: {
+            device_type: "UNKNOWN",
+            email: credentials.email,
+            password: credentials.password
+        },
+        headers: {
+            "Content-Type": "application/json",
+            "Accept": "application/json",
+            "Accept-Language": "en-US"
         }
-    })
-        .then(handleError)
-        .then(createSession);
+    }).then(createSession);
 }
 
-function handleError(response){
-    if(response.msg === 'OK'){
-        return response;
-    } else {
-        throw new Error(response.msg);
+function refreshToken(refresh_token = session.refresh_token){
+    if (refresh_token == undefined) {
+      return login();
     }
+
+    return request({
+        url: '/auth/v1/token/refresh',
+        json: {
+            refresh_token: refresh_token
+        },
+        headers: {
+            "Content-Type": "application/json",
+            "Accept": "application/json",
+            "Accept-Language": "en-US"
+        }
+    }).then(updateSession);
 }
 
-function createSession(user){
-    session.user = user
-    return user;
+function createSession(login){
+    session.user_id = login.startup_data.user.user_id;
+    session.access_token = login.access_token;
+    session.refresh_token = login.refresh_token;
+    return login;
 }
 
-function listFavoriteBusinesses(user = session.user){
+function updateSession(token){
+    session.access_token = token.access_token;
+    return token;
+}
+
+function listFavoriteBusinesses(user_id = session.user_id, access_token = session.access_token){
     return request({
-        url: '/list_favorite_businessesv5',
-        form: {
-            user_id: user.user_id,
-            user_token: user.user_token
-        }
-    }).then(handleError);
-}
\ No newline at end of file
+        url: '/item/v3/',
+        json: {
+            favorites_only: true,
+            origin: {
```

Comment:
Using a hardcoded location seems fishy.
Doesn't it work without specifying the location?

Reply:
The endpoint requires a location and will output an error if it is not set. For our use case, the location is irrelevant, since we only query our favorites. If this attribute is set to false, the location is used to return the nearest businesses around you. I understand that this is fishy and would prefer a solution without a location, but we need a dummy location for the request. If you have a nicer solution – welcome.

Reply:
Ok, this is fine then.